File: demo_unity/Assets/Scripts/LetoFeedDemo.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;
using UnityEngine.SceneManagement;
using LetoAd;

public class LetoFeedDemo : MonoBehaviour
{
    private ILTFeedCSSDK _letoSdk;
    private int _adId = 1;

    void Awake() {
        _letoSdk = new LTFeedCSSDK();
        _letoSdk.setAdListener(new EventListener());
    }

    public void OnBackClicked() {
        SceneManager.LoadScene("LetoMainDemo");
    }

    public void OnLoadClicked() {
        _letoSdk.load(_adId);
    }

    public void OnShowClicked() {
        _letoSdk.show(_adId);
    }

    public void OnHideClicked() {
        _letoSdk.hide(_adId);
    }

    public void OnDestroyClicked() {
        _letoSdk.destroy(_adId);
    }

    class EventListener : ILTFeedListener {
        public void onFeedLoaded(int adId, LTAdInfo adInfo) {
            Debug.Log("onFeedLoaded: " + adId);
        }

        public void onFeedFailed(int adId, string errMsg) {
            Debug.Log("onFeedFailed: " + adId);
        }

        public void onFeedShow(int adId, LTAdInfo adInfo) {
            Debug.Log("onFeedShow: " + adId);
        }

        public void onFeedHide(int adId, LTAdInfo adInfo) {
            Debug.Log("onFeedHide: " + adId);
        }

        public void onFeedClick(int adId, LTAdInfo adInfo) {
            Debug.Log("onFeedClick: " + adId);
        }

        public void onFeedClose(int adId, LTAdInfo adInfo) {
            Debug.Log("onFeedClose: " + adId);
        }
    }
}

File: demo_laya_js/src/ui/LetoFeedDemo.js
export default class LetoFeedDemo extends Laya.Scene {
    constructor() {
        super();
        
        //加载场景文件
        this.loadScene("test/LetoFeedDemo.scene");

        // event
        this.btnBack.on(Laya.Event.CLICK, this, this.onBackClicked)
        this.btnLoad.on(Laya.Event.CLICK, this, this.onLoadClicked)
        this.btnShow.on(Laya.Event.CLICK, this, this.onShowClicked)
        this.btnHide.on(Laya.Event.CLICK, this, this.onHideClicked)
        this.btnDestroy.on(Laya.Event.CLICK, this, this.onDestroyClicked)
    }

    adId() {
        return 1
    }

    onAwake() {
        LTFeedJSSDK.setAdListener(this);
    }

    onBackClicked() {
        Laya.Scene.open('test/LetoMainDemo.scene')
    }

    onLoadClicked() {
        LTFeedJSSDK.load(this.adId());
    }

    onShowClicked() {
        LTFeedJSSDK.show(this.adId());
    }

    onHideClicked() {
        LTFeedJSSDK.hide(this.adId());
    }

    onDestroyClicked() {
        LTFeedJSSDK.destroy(this.adId())
    }

    //Callbacks
    onFeedAdLoaded(adId, adInfo) {
        LTJSSDK.printLog(`LetoFeedDemo::onFeedAdLoaded(${adId}, ${JSON.stringify(adInfo)})`);
    }

    onFeedAdFailed(adId, errMsg) {
      LTJSSDK.printLog("LetoFeedDemo::onFeedAdFailed(" + adId + ", " + errMsg + ")");
    }

    onFeedAdShow(adId, adInfo) {
        LTJSSDK.printLog(`LetoFeedDemo::onFeedAdShow(${adId}, ${JSON.stringify(adInfo)})`);
    }

    onFeedAdHide(adId, adInfo) {
        LTJSSDK.printLog(`LetoFeedDemo::onFeedAdHide(${adId}, ${JSON.stringify(adInfo)})`);
    }

    onFeedAdClick(adId, adInfo) {
        LTJSSDK.printLog(`LetoFeedDemo::onFeedAdClick(${adId}, ${JSON.stringify(adInfo)})`);
    }

    onFeedAdClose(adId, adInfo) {
        LTJSSDK.printLog(`LetoFeedDemo::onFeedAdClose(${adId}, ${JSON.stringify(adInfo)})`);
    }
}